src/include/rbd/librbd.h

```diff
@@ -260,7 +260,7 @@ CEPH_RBD_API int rbd_rename(rados_ioctx_t src_io_ctx, const char *srcname,
                             const char *destname);
 
 CEPH_RBD_API int rbd_trash_move(rados_ioctx_t io, const char *name,
-                                uint64_t delay);
```

Comment:
Since the API is released, we cannot create a breaking change. The human-readable parsing can probably be limited to the rbd CLI since any API users will have no issue with supplying a derived time in seconds.

Reply:
I still think this comment is valid -- from an API perspective, it doesn't make much sense to have a negative delay. If some uses the rbd CLI to specify a time in the past, it should be warned/prohibited.

Reply:
In our use cases we need to set the expiration times to the past for test purposes, debugging and averting mistakes that cost space, we don't like to have unexpected things. In the end, a date is passed to the trashspec, so why do we have to use an offset as an argument instead of just passing the desired date as an absolute value? Because now to set a protected until date for an image, I have to get the date, subtract from now, pass the offset to the function so that it adds the offset from now to store the date that I initially had. With having the "uint64_t delay" as "time_t protected_until", I could just put "time(NULL)+5" in C or "int(time.time())+5" in Python2 and have the same effect. This positive offset has really no flexibility and serves a narrow variety of use cases.

Reply:
The issue is that we cannot change a released API -- so now such a change would require a ```rbd_trash_move2(..., bool expire_time_valid, time_t expire_time)```-like method. The reason that an offset delay was chosen was it solved the issue of specifying a time and also indicating whether or not a "protection period" was required (i.e. any value greater than zero).